### ISSUE TITLE
23.1 - Vectorized COPY FROM - index out of range [-1]

### ISSUE THREAD

**Issue by MT-Jacobs** [cockroachdb/cockroach#106913]
**Describe the problem**

It appears that the [23.1 upgrade](https://www.cockroachlabs.com/docs/releases/v23.1#v23-1-0-alpha-8-sql-language-changes) broke COPY FROM queries in certain instances.

We're running on Cockroach Cloud and recently ran the upgrade.

**To Reproduce**

I had a coworker attempt to run the following query from a Python script:

```python
sql=f"""COPY {self.cockroachdb_schema_name}.{self.cockroachdb_table_name} FROM STDIN WITH CSV HEADER DELIMITER '{self.delimiter}'""",
                        file=fname,
```

After running it, they received the error:

```
unexpected error from the vectorized engine: runtime error: index out of range [-1]
```

**Expected behavior**

In the previous CRDB version, this query ran without issue. It's run as a nightly script.

**Additional data / screenshots**

The query is included above, albeit in Python. The schema is pretty straightforward (with names changed):

```sql
create table my_table
(
    id               uuid not null
        constraint pk_name
            primary key,
    thing_id uuid not null
        unique,
    number_one   numeric(19, 2),
    number_two        numeric(19, 2)
);
```

**Environment:**
 - CockroachDB version [e.g. 23.1]
 - Server OS: Not sure, it's running on Cockroach Cloud using Google Cloud.
 - Client app: psycopg2

**Additional context**

Currently our nightly script to populate some analytics data isn't working because of this. We're trying out setting the `vectorize` session variable to `false` as a workaround.

---

If you need any additional details, please let me know!

Jira issue: CRDB-29763

**Comment by rytaft** [comment]
Thanks for the report, @MT-Jacobs! Sorry that you're running into this.

@cucaroach can you please take a look?

**Comment by cucaroach** [comment]
Thanks for the report @MT-Jacobs, this will be fixed in the next 23.1 dot release, should be 23.1.6.
